azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -71,6 +93,7 @@
     interface Definition<ParentT> extends
         DefinitionStages.Blank<ParentT>,
         DefinitionStages.WithAddressPrefix<ParentT>,
+        DefinitionStages.WithNetworkSecurityGroup<ParentT>,
         DefinitionStages.WithAttach<ParentT> {
     }
 
```

Comment:
I think you've got a good point, so overall I'm sold on your proposed way of organizing the interfaces.
I'm just still trying to think of this as also an opportunity to have better interface naming. I'm not really a fan of WithCreate/WithAttach. 
"*Creatable" and "*Attachable" sounded better to me but now that "Definition" should be stripped off from previous "DefinitionCreatable", these would conflict with the same named base interfaces so should not be used in that form. 
Still looking for a better naming convention though...